Flavours/stellio-flavour/common_vsilo_functionality.py

```diff
@@ -0,0 +1,677 @@
+                # and he @@ separator is just to visually separate, not to call a string.split()
+                # The prefixed vthingid is used to group and fast-get just the elements of
+                # one specific vthingid, because the leveldb hashmap is ordered lexicographycally, and
+                # we can jump to a set of keys which begin_with() something.
+                key = (v_thing_id + "@@" + entity['id']).encode()
+                value = entity['id'].encode()
+                leveldb.put(key, value)
+            else:
+                print("Exception while REST ADD of Entity " + entity['id'] + " under vThing " + v_thing_id)
+    else:
+        print(" DATA HAS ARRIVED FOR NON ACTIVE vThing " + v_thing_id + ". Skipping it.")
+
+    # return ok in any case
+    return 'OK'
+# ========== END HASHMAP + BROKER
+
+
+def handler(signal, frame):
+    print("HANDLING")
+    clean_close()
+
+
+def clean_close():
+    global connected_clients
+    connected_clients = 0
+    mqtt_data_client.disconnect()
+    mqtt_control_client.disconnect()
+    leveldb.close()
+    sys.exit()
+
+
+def start_silo_controller(broker_specific_module_name):
+    global v_silo_prefix
+    global v_thing_prefix
+    global in_control_suffix
+    global out_control_suffix
+    global out_data_suffix
+    global in_vsilo_control_topic
+    global out_vsilo_control_topic
+    global out_generic_thingvisor_control_topic
+    global mqtt_control_client
+    global mqtt_data_client
+    global connected_clients
+    global leveldb
+    global db_IP
+    global db_port
+    global db_name
+    global v_thing_collection
+    global thing_visor_collection
+    global v_silo_id
+    global brokerspecific
+    global tenant_id
+    global is_this_vsilo_systemvsilo
+    global controllerurl
+    global adminpassword
+    global flavour_params
+
+    # lets import the silo-controller-specific functions and bind them into us
+    print("importing module: " + broker_specific_module_name)
+    brokerspecific = import_module(broker_specific_module_name)
+
+    signal.signal(signal.SIGINT, handler)
+
+    # connect to the on-disk hashmap
+    shutil.rmtree('plyvelhashmap', ignore_errors=True)
+    leveldb = plyvel.DB('plyvelhashmap', create_if_missing=True)
+
+    # If you run this silo controller MANUALLY for DEBUG PURPOSES,
+    # then you may access the -d parameter to set a local address of the mongo DB
+    # and a local MQTT. This parameter is never visible when this silo
+    # is run via docker automatism (or similar) via the master controller.
+    parser = argparse.ArgumentParser(description='A vSilo Controller.')
+    parser.add_argument('-d', '--debug_mongo_ip', dest='debug_mongo_ip', help='run in debug mode and use the given IP address of the system MongoDB (usually 172.17.0.2)')
+    args = parser.parse_args()
+    # flavour_params must be a string
+    flavour_params = ""
```

Comment:
I added this line to fix this:

`NameError: name 'flavour_params' is not defined`